crates/miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -761,6 +834,72 @@ export.output_note_get_assets_info
     # => [ASSETS_COMMITMENT, num_assets, pad(11)]
 end
 
+#! Returns the recipient of the output note with the specified index.
+#!
+#! Inputs:  [note_index, pad(15)]
+#! Outputs: [RECIPIENT, pad(12)]
+#!
+#! Where:
+#! - note_index is the index of the output note whose recipient should be returned.
+#! - RECIPIENT is the commitment to the output note's script, inputs, the serial number.
+#!
+#! Panics if:
+#! - the note index is greater or equal to the total number of output notes.
+#!
+#! Invocation: dynexec
+export.output_note_get_recipient
+    # assert that the provided note index is less than the total number of output notes
+    exec.output_note::assert_note_index_in_bounds
+    # => [note_index, pad(15)]
+
+    # get the note data pointer by the provided index
+    exec.memory::get_output_note_ptr
+    # => [note_ptr, pad(15)]
+
+    # get the recipient
+    exec.memory::get_output_note_recipient
+    # => [RECIPIENT, pad(15)]
+
+    swapw drop drop drop movdn.4
+
+    # truncate the stack
+    swapw drop drop drop movdn.4
+    # => [RECIPIENT, pad(12)]
+end
+
+#! Returns the metadata of the output note with the specified index.
+#!
+#! Inputs:  [note_index, pad(15)]
+#! Outputs: [METADATA, pad(12)]
+#!
+#! Where:
+#! - note_index is the index of the output note whose metadata should be returned.
+#! - METADATA is the metadata of the output note.
+#!
+#! Panics if:
+#! - the note index is greater or equal to the total number of output notes.
+#!
+#! Invocation: dynexec
+export.output_note_get_metadata
+    # assert that the provided note index is less than the total number of output notes
+    exec.output_note::assert_note_index_in_bounds
+    # => [note_index, pad(15)]
+    
+    # get the note data pointer by the provided index
+    exec.memory::get_output_note_ptr
+    # => [note_ptr, pad(15)]
+
+    # get the metadata
+    exec.memory::get_output_note_metadata
+    # => [METADATA, pad(15)]
+
+    swapw drop drop drop movdn.4
+
+    # truncate the stack
+    swapw drop drop drop movdn.4
+    # => [METADATA, pad(12)]
```

Comment:
Seems like we are truncating the stack twice here (same in other places).

Reply:
My bad, excuse me. I still haven't woken up 🫠